custom/lib_portal/models/stock_scrap.py

```diff
@@ -0,0 +1,18 @@
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError,Warning
+
+class StockScrap(models.Model):
+    _inherit = 'stock.scrap'
+    
+    def do_scrap(self):
+        for scrap in self:
+            raise Warning("Are you sure to scrap the %s?"%scrap.product_id)
+        obj = super (StockScrap, self) .do_scrap () 
+        return obj
```

Comment:
thiese lines are never going to be executed since you are raising a warning without any condition

Reply:
this is wrong commit.I modified the code but committed it wrongly..sry for the inconvenience.Will commit the right one